### PR TITLE
Implement bundle_transactions_t object renew function

### PR DESCRIPTION
**Pull request by howjmay**
Renewing bundle transaction object is used in MAM and MAM
related implementation. This bundle renew function would
enhance the readability of the codes and make them more clear.

### CODE REVIEW

File: mam/api/tests/test_api.c [hunk 1]
```diff
@@ -540,8 +540,7 @@ static void test_api_trust() {
   free(payload);
   payload = NULL;
 
-  bundle_transactions_free(&bundle);
-  bundle_transactions_new(&bundle);
+  bundle_transactions_renew(&bundle);
```
Comment (oopsmonk) [inline mam/api/tests/test_api.c]: `bundle_transactions_t` is an UT_array you can reuse it by calling `utarray_clear`.

**Review by oopsmonk — CHANGES_REQUESTED** [review]

File: mam/api/tests/test_api.c (hunk 1 above)
Comment (howjmay) [reply mam/api/tests/test_api.c]: Let me change it into `utarray_clear` here

Comment (howjmay) [reply mam/api/tests/test_api.c]: I wonder, is it better to use `utarray_renew` here?

Comment (oopsmonk) [reply mam/api/tests/test_api.c]: No, the `bundle` cannot be null.

File: common/model/bundle.h [hunk 2]
```diff
@@ -96,6 +96,13 @@ static inline size_t bundle_transactions_size(bundle_transactions_t const *const
   return utarray_len(bundle);
 }
 
+/**
+ * @brief Re-initialize the bundle_transactions_t object.
+ *
+ * @param[in] bundle The pointer of bundle object.
+ */
+static inline void bundle_transactions_renew(bundle_transactions_t *bundle) { utarray_clear(bundle); }
```
Comment (oopsmonk) [inline common/model/bundle.h]: Please rephrase the doc as well.

```suggestion
static inline void bundle_transactions_clear(bundle_transactions_t *bundle) { utarray_clear(bundle); }
```

**Review by oopsmonk — CHANGES_REQUESTED** [review]

**Review by oopsmonk — APPROVED** [review]

**Comment by oopsmonk** [comment]
CI failed due to the out-dated toolchain, waiting for #1454 to be merged.

**Pull request merged**